### PR TITLE
MMCA-4800 | Use customs-data-store routes for email retrieval on customs-financials-frontend

### PR DESCRIPTION
**Pull request by Neozxc**
Task:
- Modify customs-financials-frontend repo so that it retrieves email addresses using the customs-data-store instead of the customs-financials-api.
- Add or amend unit tests.

### CODE REVIEW

**Review by AdamCoops — APPROVED** [review]

File: app/connectors/CustomsFinancialsApiConnector.scala [hunk 1]
```diff
@@ -33,23 +32,23 @@ class CustomsFinancialsApiConnector @Inject()(appConfig: AppConfig,
                                               metricsReporter: MetricsReporterService)(implicit ec: ExecutionContext) {
 
   def isEmailVerified(implicit hc: HeaderCarrier): Future[EmailVerifiedResponse] = {
-    httpClient.get(url"${appConfig.customsFinancialsApi}/subscriptions/subscriptionsdisplay")
+    httpClient.get(url"${appConfig.customsDataStore}/subscriptions/subscriptionsdisplay")
```
Comment (tyagikumarvineet) [inline app/connectors/CustomsFinancialsApiConnector.scala:35]: Could you please create a new class called DataStoreConnector or CustomsDataStoreConnector and move these endpoints in that class cause this is incorrect to call data store endpoints from FinancialsApiConnector? Also move the tests as well. Then call these methods from DataStoreConnector in EmailController.ApiConnector should be used, only to connect to Financials Api endpoints.

Comment (Neozxc) [reply app/connectors/CustomsFinancialsApiConnector.scala]: Initially was considering this approach at first but kept it simple. Thought perhaps if **deleteNotification** method would also need to be called by **customs-data-store** in the future, then we can just rename the files from "CustomsFinancialApi" to "CustomsDataStore" without having separate files. But this works too thanks for the feedback 👍

Comment (tyagikumarvineet) [reply app/connectors/CustomsFinancialsApiConnector.scala]: okay, deleteNotification may not be moved as it is specific to CDS financials. We moved the email address retrieval to data store as email retrieval is being used now by other CDS teams and custom data store is the one common place for other CDS teams

File: app/connectors/CustomsDataStoreConnector.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+  def isEmailVerified(implicit hc: HeaderCarrier): Future[EmailVerifiedResponse] = {
+    httpClient.get(url"${appConfig.customsDataStore}/subscriptions/subscriptionsdisplay")
+      .execute[EmailVerifiedResponse]
+      .flatMap {
+        response => Future.successful(response)
+      }
+  }
+
+  def getEmailAddress(implicit hc: HeaderCarrier): Future[EmailVerifiedResponse] = {
```
Comment (tyagikumarvineet) [inline app/connectors/CustomsDataStoreConnector.scala:39]: Could you please add a test case for this?

Comment (tyagikumarvineet) [reply app/connectors/CustomsDataStoreConnector.scala:39]: Test case has been added. Hence resolving

**Review by tyagikumarvineet — APPROVED** [review]

**Pull request merged**